app/connection/NetworkMonitor.js

```diff
@@ -0,0 +1,87 @@
+import { PendingPromise } from '@binary-com/smartcharts'; // eslint-disable-line import/no-extraneous-dependencies,import/no-unresolved
+
+class NetworkMonitor {
+    static _instance;
+
+    status_config = {
+        online: { class: 'online', tooltip: 'Online' },
+        offline: { class: 'offline', tooltip: 'Offline' },
+        blinking: { class: 'blinker', tooltip: 'Connecting to server' },
+    };
+    last_status;
+    last_is_online;
+    statusStarted = false;
+    statusStartedPromise = new PendingPromise();
+
+    setNetworkStatus = (status) => {
+        const is_online = this.isOnline();
+        if (status !== this.last_status || is_online !== this.last_is_online) {
+            this.last_status = status;
+            this.last_is_online = is_online;
+
+            if (typeof this.onStatusUpdated === 'function') this.onStatusUpdated(this.status_config[status], is_online);
```

Comment:
I think You can use the new optional chaining operator instead of this if statement:

```suggestion
            this.onStatusUpdated?.(this.status_config[status], is_online);
```

Reply:
@mahdiaryayi-fs mentioned above